### ISSUE TITLE
CDN Releases

### ISSUE THREAD

**Issue by boredom2** [google/shaka-player#1250]
Hi there,

the 2.3.1 Builld seems to have Problems on the CDN Releases.
On CDNJS, there is no Build at all (404) and on Google, the Version that is published there under "2.3.1" has the Notice "2.3.1-dirty" in the Code itself. Is that useable or what means the "dirty" in this Case?

Thanks!

**Comment by vaage** [comment]
@boredom2 Thank you for finding this. 

I was able to confirm that all the links on https://cdnjs.com/libraries/shaka-player for version 2.3.1 all return 404.

I was able to confirm that the version string on the Google CDN is "v2.3.1-dirty".

Since @joeyparrish handles all our releases, I will talk with him about this. I'll label this as a bug.

**Comment by jimaek** [comment]
Btw you could also use jsDelivr. It pulls files directly from npm, so no need to do anything at all. 
https://cdn.jsdelivr.net/npm/shaka-player/
https://cdn.jsdelivr.net/npm/shaka-player
https://cdn.jsdelivr.net/npm/shaka-player@2.3.1/dist/shaka-player.compiled.js

**Comment by joeyparrish** [comment]
The broken links at cdnjs seems to be a cdnjs bug.  They pull our stuff automatically from npm.

Doing anything with cdnjs yourself is challening, though, because they keep all libraries & versions in git, in one repo.  Their git repo is so massive it typically fails to check out.  They use a bot to pull new releases from npm and update the repo.

@boredom2, I recommend you file a bug against cdnjs about the broken links.  In the mean time, there are other options, such as [jsDelivr](https://cdn.jsdelivr.net/npm/shaka-player/) and [Google's Hosted Libraries](https://developers.google.com/speed/libraries/#shaka-player).

I will keep this open to investigate why the version number in our official v2.3.1 release claims to be dirty (local modifications), and what local mods there were, if any, that affected the build.

**Comment by boredom2** [comment]
Thanks, I didnt know about jsDelivr - copied :)

**Comment by joeyparrish** [comment]
The build in NPM is fine, and has version string "v2.3.1", as it should.  The build on Google's Hosted Libraries seems to be the only one that says "v2.3.1-dirty".  Still tracking down what the dirt is.

**Comment by joeyparrish** [comment]
The last several releases are fine: v2.2.9, v2.2.10, v2.3.0.  The only difference between the npm build of v2.3.1 and the Hosted Libraries build of v2.3.1 is the version string, in which the latter holds the "-dirty" suffix.

The "-dirty" suffix can happen if there are any local mods anywhere in the tree, including in the docs or the build scripts.  I am still tracking down what part of the release process caused this suffix, but the builds are all correct and safe to use.

**Comment by boredom2** [comment]
We put the 2.3.1 Builds to Production already and i can confirm - no Issues there:-)

**Comment by joeyparrish** [comment]
The "-dirty" suffix is caused by b3b7860f8cd2d57e4649fcffea7c016a4bd81501, in which JS files all get LF line endings.  The file `third_party/jsdoc/node_modules/catharsis/node_modules/underscore-contrib/underscore.array.selectors.js` has CRLF endings, and these were converted to LF in the release process for Google Hosted Libraries.  This constituted a change in the working directory, which led to the "-dirty" flag, even though no changes were made to the library.

So, to summarize all of the above, this is a cosmetic issue only, only affects the copy in Google Hosted Libraries, and the process will be fixed for the v2.3.2 release.
